src/client/src/api/fetchApiV2.js

```diff
@@ -36,9 +37,26 @@ export async function fetchLayersByProfileId(profileId) {
   return await fetchApiV2(`layer?profileId=${profileId}`, "GET");
 }
 
+export const fetchAllCodeLists = async () => {
+  return await fetchApiV2(`codelist`, "GET");
+};
+
+export const updateCodeLists = async codelist => {
+  return await fetchApiV2("codelist", "PUT", codelist);
+};
+
 export async function updateLayer(layer) {
   // remove derived objects
   delete layer.createdBy;
   delete layer.updatedBy;
   return await fetchApiV2(`layer`, "PUT", layer);
 }
+
+// Enable using react-query outputs across the application.
+
+// eslint-disable-next-line react-hooks/rules-of-hooks
+export const useDomains = () =>
+  useQuery("domains", async () => {
+    const domains = fetchApiV2(`codelist`, "GET");
+    return domains;
```

Comment:
Q: Brauchts hier kein `await`? zB
```
return await fetchApiV2(`codelist`, "GET");
```

Reply:
Nein react-query arbeitet mit Promises die dann erst beim konsumieren verarbeitet werden. Bisschen wie observables in Angular falls du das noch kannst :D